airbyte-cdk/bulk/toolkits/load-iceberg-parquet/src/main/kotlin/io/airbyte/cdk/load/data/iceberg_parquet/AirbyteTypeToIcebergSchema.kt

```diff
@@ -0,0 +1,109 @@
+ */
+
+package io.airbyte.cdk.load.data.iceberg_parquet
+
+import io.airbyte.cdk.load.data.AirbyteType
+import io.airbyte.cdk.load.data.ArrayType
+import io.airbyte.cdk.load.data.ArrayTypeWithoutSchema
+import io.airbyte.cdk.load.data.BooleanType
+import io.airbyte.cdk.load.data.DateType
+import io.airbyte.cdk.load.data.IntegerType
+import io.airbyte.cdk.load.data.NumberType
+import io.airbyte.cdk.load.data.ObjectType
+import io.airbyte.cdk.load.data.ObjectTypeWithEmptySchema
+import io.airbyte.cdk.load.data.ObjectTypeWithoutSchema
+import io.airbyte.cdk.load.data.StringType
+import io.airbyte.cdk.load.data.TimeTypeWithTimezone
+import io.airbyte.cdk.load.data.TimeTypeWithoutTimezone
+import io.airbyte.cdk.load.data.TimestampTypeWithTimezone
+import io.airbyte.cdk.load.data.TimestampTypeWithoutTimezone
+import io.airbyte.cdk.load.data.UnionType
+import io.airbyte.cdk.load.data.UnknownType
+import java.util.UUID
+import org.apache.iceberg.Schema
+import org.apache.iceberg.types.Type
+import org.apache.iceberg.types.Types
+import org.apache.iceberg.types.Types.NestedField
+
+class AirbyteTypeToIcebergSchema {
+
+    fun convert(airbyteSchema: AirbyteType): Type {
+        return when (airbyteSchema) {
+            is ObjectType -> {
+                Types.StructType.of(
+                    *airbyteSchema.properties.entries
+                        .map { (name, field) ->
+                            if (field.nullable) {
+                                NestedField.optional(
+                                    UUID.randomUUID().hashCode(),
+                                    name,
+                                    convert(field.type)
+                                )
+                            } else {
+                                NestedField.required(
+                                    UUID.randomUUID().hashCode(),
+                                    name,
+                                    convert(field.type)
+                                )
+                            }
+                        }
+                        .toTypedArray()
+                )
+            }
+            is ArrayType -> {
+                val convert = convert(airbyteSchema.items.type)
+                if (airbyteSchema.items.nullable) {
+                    return Types.ListType.ofOptional(UUID.randomUUID().hashCode(), convert)
+                }
+                return Types.ListType.ofRequired(UUID.randomUUID().hashCode(), convert)
+            }
+            is ArrayTypeWithoutSchema ->
+                throw IllegalArgumentException("Array type without schema is not supported")
+            is BooleanType -> Types.BooleanType.get()
+            is DateType -> Types.DateType.get()
+            is IntegerType -> Types.LongType.get()
+            is NumberType -> Types.DoubleType.get()
+            is ObjectTypeWithEmptySchema ->
+                throw IllegalArgumentException("Object type with empty schema is not supported")
```

Comment:
Do we want to blow up here or just log that we don't support it and skip this field?  Same with the `ObjectTypeWithoutSchema`.

Reply:
@subodh1810 I assume we're relying on an upstream mapper to remove these types? (similar to https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/bulk/toolkits/load-avro/src/main/kotlin/io/airbyte/cdk/load/data/avro/AvroMapperPipelineFactory.kt#L23 )

in which case it's a legit bug if we get these types here, and throwing an exception seems right

Reply:
Yup